### PR TITLE
KOA-4897 Add rounding property on star rating component

### PR DESCRIPTION
**Pull request by marianeum**
Added a new rounding property for rounding down (default), up or to nearest, same as web.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: docs/view/StarRating/README.md [hunk 1]
```diff
@@ -22,13 +23,24 @@ Example of a Star Rating in Kotlin
...
+import net.skyscanner.backpack.starrating.RoundingType
 
 BpkStarRating(context).apply {
   maxRating = 5
   rating = 2.5f
+  rounding = RoundingType.UP
 }
 ```
 
+# Props
```
Comment (marianeum) [inline docs/view/StarRating/README.md:35]: I saw this on the web docs and thought it'd be very helpful for this component particularly to know the properties & default values

File: Backpack/src/main/java/net/skyscanner/backpack/starrating/RoundingType.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.starrating
+
+enum class RoundingType {
+  DOWN,
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/starrating/RoundingType.kt:22]: We mostly use CamelCase in Backpack for enum values.

File: Backpack/src/main/java/net/skyscanner/backpack/starrating/internal/BpkStarRatingBase.kt [hunk 3]
```diff
@@ -85,6 +97,7 @@ open class BpkStarRatingBase internal constructor(
     ).use {
       _maxRating = it.getInt(R.styleable.BpkStarRating_maxRating, maxRating)
       _rating = it.getFloat(R.styleable.BpkStarRating_rating, maxRating / 2f)
+      _rounding = it.getInt(R.styleable.BpkStarRating_rounding, rounding.ordinal).let { id -> RoundingType.values()[id] }
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/starrating/internal/BpkStarRatingBase.kt:100]: We don't usually rely on ordinals and use int constants instead – this gives us more flexiblity when modifying enums. You can see an example in BpkBadge, for instance.

File: app/src/androidTest/java/net/skyscanner/backpack/starrating/BpkStarRatingTest.kt [hunk 4]
```diff
@@ -35,71 +35,99 @@ class BpkStarRatingTest : BpkSnapshotTest() {
...
 
   @Test
-  fun screenshotTestStarRating_Default() {
+  fun default() {
     snap(rating)
   }
 
   @Test
-  fun screenshotTestStarRating_CustomMaxRatingIncreasing() {
+  fun customMaxRatingIncreasing() {
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/starrating/BpkStarRatingTest.kt:43]: 👍

**Review by bvitaliyg — APPROVED** [review]

**Pull request merged**